docs/writing-tests/unit-testing.md

```diff
@@ -0,0 +1,137 @@
+---
+title: 'Unit tests'
+---
+
+Teams test a variety of UI characteristics using different tools. Each tool requires you to replicate the same component state over and over. That’s a maintenance headache. Ideally, you’d set up your tests in the same way and reuse that across tools.
+
+Storybook enables you to isolate a component and capture its use cases in a `*.stories.js|ts` file. Stories are standard JavaScript modules cross-compatible with the whole JavaScript ecosystem—no API lock-in.
+
+Stories are a practical starting point for UI testing. Import stories into tools like [Jest](https://jestjs.io/), [Testing Library](https://testing-library.com/), [Puppeteer](https://pptr.dev/), to save time and maintenance work.
+
+## Write a test with Testing Library
+
+[Testing Library](https://testing-library.com/) is a suite of helper libraries for browser-based interaction tests. With [Component Story Format](../api/csf.md), your stories are reusable with Testing Library. Each named export (story) is renderable within your testing setup. For example, if you were working on a login component and wanted to test the invalid credentials scenario, here's how you could write your test:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'react/component-test-with-testing-library.js.mdx',
+    'react/component-test-with-testing-library.ts.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+<div class="aside">
+
+💡 You can use Testing Library out-of-the-box with [Storybook Interaction Testing](./interaction-testing.md).
+
+</div>
+
+Once the test runs, it loads the story and renders it. [Testing Library](https://testing-library.com/) then emulates the user's behavior and checks if the component state has been updated.
+
+## Configure
+
+By default, Storybook offers a zero-config setup for React and other frameworks via addons, allowing you to run your stories as tests with Testing Library. However, adding a configuration file can extend your test setup and include additional global [decorators](../writing-stories/decorators.md#global-decorators), [parameters](../writing-stories/parameters.md#global-parameters), and other features. To do so, create a `setup.js|ts` file as follows:
```

Comment:
The point of `setProjectAnnotations` is to pass the annotations which reside normally in `preview.js` which are global and processed automatically by Storybook, however because users are running their stories outside of Storybook, they need to do it themselves. It's not necessarily to just "extend" the setup – it's more like "you have to do this if you have any important annotation in your preview.js file such as decorators and parameters"